farmdata2_modules/fd2_tabs/resources/RegexInputComponent.js

```diff
@@ -97,43 +97,51 @@ let RegexInputComponent = {
           // Handles validation of the input value
           blurEventHandler: function (e) {
             const inputVal = e.target.value;
-            this.validateVal(inputVal)
-            
+            this.validateVal(inputVal) 
         },
 
-        // This click even handler points the event to the input box to address
-        // an issue in FireFox where clicking on the increment/decrement buttons 
-        // does not actually focus the input box.
-        clickEventHandler: function (e) {
-            e.target.focus()
-        },
+          // This click even handler points the event to the input box to address
+          // an issue in FireFox where clicking on the increment/decrement buttons 
+          // does not actually focus the input box.
+          clickEventHandler: function (e) {
+              e.target.focus()
+          },
 
-        validateVal(isValid){
-          const re = new RegExp(this.regExp)
-          this.isMatch = re.test(isValid)
-          this.updateColor(this.isMatch)
-          this.$emit('match-changed', this.isMatch)
-        },
+          validateVal(inputVal){
+            const re = new RegExp(this.regExp)
+            const temp = this.isMatch
+            this.isMatch = re.test(inputVal)
+            this.$emit('input-changed', inputVal)
+            this.updateColor(this.isMatch)
+            if(this.isMatch != temp){
+              this.$emit('match-changed', this.isMatch)
+            }
+          },
 
-        updateColor(setNewColor){
-          if(!setNewColor){
-            this.inputStyle.backgroundColor = this.setColor
-        }
-        else{
-            this.inputStyle.backgroundColor = 'white'
-        }
+          updateColor(setNewColor){
```

Comment:
Change parameter to be more reflective of what's being passed "isValid", "isMatched", "matches", etc.